apis/python/tests/test_indexed_dataframe.py

```diff
@@ -289,6 +294,267 @@ def make_dataframe(request):
 )
 def test_index_types(tmp_path, make_dataframe):
     """Verify that the index columns can be of various types"""
-    sdf = soma.IndexedDataFrame(tmp_path.as_posix())
-    sdf.create(make_dataframe.schema, index_column_names=["index"])
-    sdf.write(make_dataframe)
+    sidf = soma.IndexedDataFrame(tmp_path.as_posix())
+    sidf.create(make_dataframe.schema, index_column_names=["index"])
+    sidf.write(make_dataframe)
+
+
+def make_multiply_indexed_dataframe(tmp_path, index_column_names: List[str]):
+    """
+    Creates a variably-indexed IndexedDataFrame for use in tests below.
+    """
+    schema = pa.schema(
```

Comment:
IMHO, this should support non-int indexing, and just xfail it for now.  That way we have an accurate representation of the gaps in the API (AKA test-driven development)

Reply:
@bkmartinjr good idea but it won't pass `typeguard`

I guess I can put non-int in the type-specs to say that string etc _are_ allowed at that level, then xfail them in the unit-test code, and do run-time assertions in `create` ...

Reply:
I'm not sure what this means other than the interface needs to be improved.  If typeguard is throwing an error, then lets fix the interface...

Can you elaborate?  Are you claiming a bug in typeguard?

Reply:
@bkmartinjr I'm saying if I typedef to say (as it is at present)
```
SparseIndexedDataFrameCoordinate = Union[
    None, Union[int, slice, Sequence[int], pa.Array, pa.ChunkedArray, np.ndarray]
                                     ^^^ right here
]
```
then any tests in which I try passing a string (say) coordinate will fail typeguard.

This isn't a bug in typeguard; it's typeguard doing it's job.

What I'll need to do is replace that that typedef with
```
SparseIndexedDataFrameCoordinate = Union[
    None, Union[int, slice, Sequence[xxx], pa.Array, pa.ChunkedArray, np.ndarray]
                                     ^^^ something more broad
```
and then do runtime checks as you suggest.

Reply:
Non-int indexing will be handled in #418 and #419.